Frontend/src/components/SignupForm.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, ChangeEvent, useRef } from "react";
+import { getBody, getResponse, resetForm } from "../helper";
+
+interface SignUpFormState {
+  name: string;
+  email: string;
+  age: string;
+  profilePicture: any;
+}
+
+const SignUpForm: React.FC = () => {
+  const [formData, setFormData] = useState<SignUpFormState>({
+    name: "",
+    email: "",
+    age: "",
+    profilePicture: null,
+  });
+  const [response, setResponse] = useState<any>();
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData((prevData) => ({ ...prevData, [name]: value }));
+  };
+
+  const handleFileUpload = (event: ChangeEvent<HTMLInputElement>) => {
+    setFormData((prevData) => ({
+      ...prevData,
+      profilePicture: event.target.files?.[0] as any,
+    }));
+  };
+
+  const HandleSubmit = async (event: any) => {
+    event.preventDefault();
+    const body = getBody(event, formData);
+    let res: any = await fetch("http://localhost:3001/saveCredentials", {
+      method: "POST",
+      body,
+    });
+    setResponse(await getResponse(res, event));
```

Comment:
Always use some variable to store data and then set that to state
const result= await getResponse(res, event)